src/did.ts

```diff
@@ -236,6 +243,18 @@ export class DID {
       }
     }
 
+    const signerDid = didResolutionResult.didDocument?.id
+    if (options.issuer && options.issuer !== signerDid) {
+      const issuerUrl = didWithTime(options.issuer, options.atTime)
```

Comment:
Maybe break this and the `timecheckEnabled` conditional into helper functions?

Reply:
TBH, I would like to accumulate a little more weirdness like this, and extract `verifyJWS` into own verification service. It clearly does not belong here as a method of `DID` object.

Reply:
Well, ideally most of this logic could move into the did-jwt package. Which btw should be renamed did-jose at some point: https://github.com/decentralized-identity/did-jwt/issues/170

Reply:
Ideally, yeah.

Reply:
I really want to accumulate weirdness here nevertheless to guide the further changes.